pkg/ingester/wal.go

```diff
@@ -180,7 +180,7 @@ func (w *walWrapper) run() {
 
 const checkpointPrefix = "checkpoint."
 
-func (w *walWrapper) performCheckpoint() (err error) {
+func (w *walWrapper) performCheckpoint(immediate bool) (err error) {
```

Comment:
With this change, there's no guarantee that the next `performCheckpoint()` (called by the priodically `CheckpointDuration` ticker) will be performed once the previous one has completed. Shouldn't we protect from this case?

Reply:
Good catch! I thought it was protected by a mutex, but apparently now. I have added one now.